runtime-template-client/runtime-template-client.go

```diff
@@ -0,0 +1,197 @@
+package runtimetemplateclient
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/astronomer/astro-cli/pkg/httputil"
+)
+
+const (
+	maxExtractSize = 100 << 20
+)
+
+type Client interface {
+	DownloadAndExtractTemplate(repoURL, branch, templateDir, destDir string) error
+}
+
+type HTTPAstroTemplateClient struct {
+	*httputil.HTTPClient
+}
+
+func NewHTTPAstroTemplateClient(client *httputil.HTTPClient) *HTTPAstroTemplateClient {
+	return &HTTPAstroTemplateClient{
+		client,
+	}
+}
+
+func (c *HTTPAstroTemplateClient) DownloadAndExtractTemplate(repoURL, templateDir, destDir string) error {
+	tarballURL := fmt.Sprintf("%s/tarball/main", repoURL)
+
+	doOpts := &httputil.DoOptions{
+		Path:   tarballURL,
+		Method: http.MethodGet,
+	}
+
+	resp, err := c.Do(doOpts)
+	if err != nil {
+		return fmt.Errorf("failed to download tarball: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("failed to download tarball, status code: %d", resp.StatusCode)
+	}
+
+	tempDir, err := os.MkdirTemp("", "extracted-")
+	if err != nil {
+		return fmt.Errorf("failed to create temp directory: %w", err)
+	}
+	defer os.RemoveAll(tempDir)
+
+	// Extract the tarball to the temporary directory
+	err = extractTarGz(resp.Body, tempDir, templateDir)
+	if err != nil {
+		return fmt.Errorf("failed to extract tarball: %w", err)
+	}
+
+	// Copy the extracted template directory to the destination directory
+	srcDir := filepath.Join(tempDir, templateDir)
+
+	if _, err := os.Stat(srcDir); os.IsNotExist(err) {
+		return fmt.Errorf("template directory %s not found", templateDir)
+	}
+	if err := copyDir(srcDir, destDir); err != nil {
+		return fmt.Errorf("failed to copy template directory: %w", err)
+	}
+
+	return nil
+}
+
+func extractTarGz(r io.Reader, dest, templateDir string) error {
```

Comment:
could we add unit test for this file logic 🙂

Reply:
Yes, I have included the tests for runtime-template-client